pexv2/pd.go

```diff
@@ -22,6 +34,98 @@ type InputDescriptor struct {
 	Constraints Constraints `json:"constraints"`
 }
 
+type tokenizedField struct {
+	token string
+	path  string
+}
+
+// SelectCredentials selects vcJWTs based on the constraints defined in the input descriptor
+func (ind InputDescriptor) SelectCredentials(vcJWTs []string) ([]string, error) {
+	jsonSchema := JSONSchema{
+		Schema:     "http://json-schema.org/draft-07/schema#",
+		Type:       "object",
+		Properties: make(map[string]Filter, len(ind.Constraints.Fields)),
+		Required:   make([]string, 0, len(ind.Constraints.Fields)),
+	}
+
+	tokenizedFields := make([]tokenizedField, 0, len(ind.Constraints.Fields))
+	tokens := make([]string, len(ind.Constraints.Fields))
+
+	for i, field := range ind.Constraints.Fields {
+		token := strconv.FormatInt(time.Now().UnixNano(), 10)
+		tokens[i] = token
+
+		for _, path := range field.Path {
+			tf := tokenizedField{token: token, path: path}
+			tokenizedFields = append(tokenizedFields, tf)
+		}
+
+		if field.Filter != nil {
+			jsonSchema.AddProperty(token, *field.Filter)
+		}
+	}
+
+	sch, err := json.Marshal(jsonSchema)
+	if err != nil {
+		return nil, fmt.Errorf("error marshalling schema: %w", err)
+	}
+
+	schema, err := jsonschema.CompileString(ind.ID, string(sch))
+	if err != nil {
+		return nil, fmt.Errorf("error compiling schema: %w", err)
+	}
+
+	matched := make([]string, 0, len(vcJWTs))
+
+	for _, vcJWT := range vcJWTs {
+		tokensFound := make(map[string]bool, len(tokenizedFields))
+
+		decoded, err := vc.Decode[vc.Claims](vcJWT)
+		if err != nil {
+			continue
+		}
+
+		var jwtPayload map[string]any
+		payload, err := base64.RawURLEncoding.DecodeString(decoded.JWT.Parts[1])
```

Comment:
i used `json.Marshal(decoded.JWT.Claims)` - i imagine `Parts[1]` is basically the base64 encoded claims? is there a benefit to deserializing it it this way?

Reply:
It's grabbing the entire vcjwt vs just claims